nuxeo-coldstorage/src/main/java/org/nuxeo/coldstorage/operations/DeleteFromcoldstorage.java

```diff
@@ -0,0 +1,57 @@
+package org.nuxeo.coldstorage.operations;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.nuxeo.coldstorage.helpers.ColdStorageHelper;
+import org.nuxeo.ecm.automation.core.Constants;
+import org.nuxeo.ecm.automation.core.annotations.Context;
+import org.nuxeo.ecm.automation.core.annotations.Operation;
+import org.nuxeo.ecm.automation.core.annotations.OperationMethod;
+import org.nuxeo.ecm.core.api.CoreSession;
+import org.nuxeo.ecm.core.api.DocumentModel;
+import org.nuxeo.ecm.core.api.NuxeoException;
+import org.nuxeo.ecm.core.api.security.SecurityConstants;
+
+import java.util.Optional;
+
+import static javax.servlet.http.HttpServletResponse.SC_INTERNAL_SERVER_ERROR;
+
+/**
+ * Delete the main content associated with the input {@link DocumentModel} from cold storage.
+ *
+ * @since 11.0
+ */
+@Operation(id = MoveToColdStorage.ID, category = Constants.CAT_BLOB, label = "Delete from Cold Storage", description = "Delete the main document content from the cold storage.")
+public class DeleteFromcoldstorage {
+
+    private static final Logger l = LogManager.getLogger(DeleteFromcoldstorage.ID);
+
+    public static final String ID = "Document.DeleteFromColdStorage";
+
+    @Context
+    protected CoreSession session;
+
+    @OperationMethod
+    public void run(DocumentModel d) {
+        if (!session.hasPermission(d.getRef(), SecurityConstants.REMOVE)) {
```

Comment:
Why a manual security check? `session.removeChildren` already does its own security checks. If this security check does something more, there should be a comment explaining why.